wpimath/src/main/java/edu/wpi/first/math/interpolation/TimeInterpolatableBuffer.java

```diff
@@ -0,0 +1,139 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.math.interpolation;
+
+import java.util.NavigableMap;
+import java.util.TreeMap;
+
+import edu.wpi.first.wpiutil.math.MathUtil;
+
+/**
+ * The TimeInterpolatableBuffer provides an easy way to estimate past measurements. One
+ * application might be in conjunction with the DifferentialDrivePoseEstimator,
+ * where knowledge of the robot pose at the time when vision or other global measurement
+ * were recorded is necessary, or for recording the past angles of mechanisms as measured
+ * by encoders.
+ *
+ * @param <T> The Interpolatable stored in this buffer.
+ */
+public class TimeInterpolatableBuffer<T> {
+
+  private final double m_historySize;
+  private final InterpolateFunction<T> m_interpolatingFunc;
+  private final NavigableMap<Double, T> m_buffer = new TreeMap<>();
+
+  private TimeInterpolatableBuffer(InterpolateFunction<T> interpolateFunction,
+                                   double historySizeSeconds) {
+    this.m_historySize = historySizeSeconds;
+    this.m_interpolatingFunc = interpolateFunction;
+  }
+
+  /**
+   * Create a new TimeInterpolatableBuffer.
+   *
+   * @param interpolateFunction The function used to interpolate between values.
+   * @param historySizeSeconds  The history size of the buffer.
+   * @param <T>                 The type of data to store in the buffer.
+   */
+  public static <T extends Interpolatable<T>> TimeInterpolatableBuffer<T> createBuffer(
+      InterpolateFunction<T> interpolateFunction, double historySizeSeconds) {
+    return new TimeInterpolatableBuffer<>(interpolateFunction, historySizeSeconds);
+  }
```

Comment:
Why is this method needed? If T extends Interpolatable, you can use the method on Line 54 since the interpolating function is already defined in the interface.

Reply:
typo, T shouldn't be interpolatable. Fixed.